db/migrate/20150525081931_remove_duplicate_tokens.rb

```diff
@@ -0,0 +1,18 @@
+class RemoveDuplicateTokens < ActiveRecord::Migration
+  def up
+    existing_tokens = Host::Managed.joins(:token).map(&:token).compact
+    if existing_tokens.empty?
+      Token.delete_all
+    else
+      Token.where('id not in (?)', existing_tokens).delete_all
+    end
+
+    remove_index :tokens, :host_id if index_exists? :tokens, :host_id
```

Comment:
I think you might need to temporarily remove the foreign key between hosts and tokens in order to do this under MySQL... also check if the foreign key exists first!